pkg/reconciler/apiserversource/apiserversource.go

```diff
@@ -338,3 +365,84 @@ func (r *Reconciler) createCloudEventAttributes(src *v1.ApiServerSource) ([]duck
 	}
 	return ceAttributes, nil
 }
+
+// createOIDCRole: this function will call resources package to get the role object
+// and then pass to kubeclient to make the actual OIDC role
+func (r *Reconciler) createOIDCRole(ctx context.Context, source *v1.ApiServerSource) error {
+	roleName := resources.GetOIDCTokenRoleName(source.Name)
+
+	//Call kubeclient and see whether the role exist or not
+	role, err := r.roleLister.Roles(source.GetNamespace()).Get(roleName)
```

Comment:
nit: can you adjust the comment as we switched to the lister and move this part above the `if apierrs.IsNotFound(err)`. Because getting an error and not checking it the first place might be confusing initially.
Then you could also rename the `errMakeRole` to simply `err`.